ports/esp32/boards/TINYPICO/modules/tinypico.py

```diff
@@ -0,0 +1,136 @@
+# and a full battery not charging - This is why the charge LED flashes
+def get_battery_charging():
+    """
+    Returns the current battery charging state.
+    This can trigger false positives as the charge IC can't tell the difference between a full battery or no battery connected.
+    """
+    measuredVal = 0                         # start our reading at 0
+    io = Pin( BAT_CHARGE, Pin.IN )          # Assign the pin to read
+  
+    for y in range(0, 10):                  # loop through 10 times adding the read values together to ensure no false positives    
+        measuredVal += io.value()
+
+    return ( measuredVal == 0 )             # return True if the value is 0
+
+# Return the internal PICO-D4 temperature in Farenheit
+def get_internal_temp_F():
+    """Return the internal PICO-D4 temperature in farenheit."""
+    return esp32.raw_temperature()
+
+# Return the internal PICO-D4 temperature in Celcius
+def get_internal_temp_C():
+    """Return the internal PICO-D4 temperature in celcius."""
+    return ( esp32.raw_temperature() - 32 ) / 1.8
+
+# Power to the on-oard Dotstar is controlled by a PNP transistor, so low is ON and high is OFF
+# We also need to set the Dotstar clock and data pins to be inputs to prevent power leakage when power is off
+# This might be improved at a future date
+# The reason we have power control for the Dotstar is that it has a quiescent current of around 1mA, so we
+# need to be able to cut power to it to minimise power consumption during deep sleep or with general battery powered use
+# to minimse un-needed battery drain
+def set_dotstar_power( state ):
+    """Set the power for the on-board Dostar to allow no current draw when not needed."""
+    # Set the power pin to the inverse of state 
+    if state:
+        Pin( DOTSTAR_PWR, Pin.OUT, None )   # Break the PULL_HOLD on the pin
+        Pin( DOTSTAR_PWR ).value( False )   # Set the pin to LOW to enable the Transistor
+    else:
+        Pin(13, Pin.IN, Pin.PULL_HOLD) # Set PULL_HOLD on the pin to allow the 3V3 pullup to work
+                   
+    Pin(DOTSTAR_CLK, Pin.OUT if state else Pin.IN )     # If power is on, set CLK to be output, otherwise input
+    Pin(DOTSTAR_DATA, Pin.OUT if state else Pin.IN )    # If power is on, set DATA to be output, otherwise input
+
+    # A small delay to let the IO change state
+    time.sleep(.035)
+
+# Dotstar rainbow colour wheel
+def dotstar_color_wheel( wheel_pos ):
+    """Color wheel to allow for cycling through the rainbow of RGB colors."""
+    wheel_pos = wheel_pos % 255
+    
+    if wheel_pos < 85:
+        return (255 - wheel_pos * 3, 0, wheel_pos * 3)   
+    elif wheel_pos < 170:
+        wheel_pos -= 85
+        return (0, wheel_pos * 3, 255 - wheel_pos * 3)
+    else:
+        wheel_pos -= 170
+        return ( wheel_pos * 3, 255 - wheel_pos * 3, 0)
+
+# Go into deep sleep but shut down the APA first to save power
+# Use this if you want lowest deep  sleep current
+def go_deepsleep( t ):
```

Comment:
Consider just `deepsleep`, avoid verbs like `do`, `calculate` etc when the noun makes the action obvious.

Reply:
Sure, I will keep that in mind for the future, thanks!
This needs to stay compatible with the 2000+ TinyPICOs out there with the non-frozen helper so will leave it for now.

Reply:
Deprecated? :yum: 
Makes sense to leave it for compatibility.